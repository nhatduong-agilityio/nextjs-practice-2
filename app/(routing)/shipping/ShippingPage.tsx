'use client'

import { Flex, Heading, Stack } from '@chakra-ui/react'
import LoadingIndicator from '@components/Common/LoadingIndicator'
import dynamic from 'next/dynamic'
import React from 'react'

const ShippingForm = dynamic(() => import('@components/Shipping/ShippingForm'), {
  loading: () => <LoadingIndicator size='lg' />,
})

const ShippingPage = () => (
  <Flex w='full' flexDir='column' p={['30px 20px', '50px 100px']}>
    <Stack flexDir='row' justifyContent='flex-start' alignItems='center' pb={['10px', '35px']}>
      <Heading variant={['extraBoldMD', 'extraBoldXL']} lineHeight='inherit'>
        Shipping Data
      </Heading>
    </Stack>

    <ShippingForm />
  </Flex>
)

export default ShippingPage
