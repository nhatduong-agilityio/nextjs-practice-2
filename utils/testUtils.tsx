import { ChakraProvider } from '@chakra-ui/react'
import { AuthProvider } from '@store/providers/AuthProvider'
import { LoadingProvider } from '@store/providers/LoadingProvider'
import type { RenderOptions, RenderResult } from '@testing-library/react'
import { render } from '@testing-library/react'
import { customTheme } from '@themes/index'
import type { ReactElement } from 'react'

const AllTheProviders = ({ children }: { children: React.ReactNode }) => {
  return (
    <ChakraProvider theme={customTheme}>
      <LoadingProvider>
        <AuthProvider>{children}</AuthProvider>
      </LoadingProvider>
    </ChakraProvider>
  )
}

const customRender = (ui: ReactElement, options?: Omit<RenderOptions, 'wrapper'>): RenderResult =>
  render(ui, { wrapper: AllTheProviders, ...options })

export * from '@testing-library/react'
export { customRender as render }
