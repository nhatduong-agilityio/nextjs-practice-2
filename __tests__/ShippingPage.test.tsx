import type { RenderResult } from '@utils/testUtils'
import { cleanup, render, waitFor } from '@utils/testUtils'
import ShippingPage from '../app/(routing)/shipping/ShippingPage'

describe('WishListPage', () => {
  global.fetch = jest.fn(() =>
    Promise.resolve({
      json: () => Promise.resolve(),
    }),
  ) as jest.Mock

  let wrapper: RenderResult

  const setup = async () => {
    const view = render(<ShippingPage />)

    await waitFor(() => {
      wrapper = view
    })
  }

  beforeEach(() => {
    cleanup()

    setup()
  })

  it('Should WishListPage match snapshot', async () => {
    const { container } = wrapper

    expect(container).toMatchSnapshot()
  })
})
