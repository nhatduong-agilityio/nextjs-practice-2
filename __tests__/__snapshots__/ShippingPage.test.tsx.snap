// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WishListPage Should WishListPage match snapshot 1`] = `
<div>
  <div
    class="css-2nkcax"
  >
    <div
      class="chakra-stack css-1tryy8i"
    >
      <h2
        class="chakra-heading css-1k4oj01"
      >
        Shipping Data
      </h2>
    </div>
    <div
      class="css-4css3g"
    >
      <div
        class="chakra-stack css-ppmla1"
        style="margin-top: 0px;"
      >
        <form>
          <div
            class="chakra-form-control css-1kxonj9"
            role="group"
          >
            <div
              class="chakra-stack css-e8q362"
            >
              <div
                class="chakra-stack css-13203fq"
                style="margin: 0px;"
              >
                <p
                  class="chakra-text css-1bmwq82"
                >
                  Name
                </p>
                <input
                  class="chakra-input css-1nkx82e"
                  id="field-:r0:"
                  name="name"
                  placeholder="Ex : Christian Tanaka"
                  type="text"
                />
              </div>
              <div
                class="chakra-stack css-13203fq"
                style="margin: 0px;"
              >
                <p
                  class="chakra-text css-1bmwq82"
                >
                  Phone Number
                </p>
                <input
                  class="chakra-input css-1nkx82e"
                  id="field-:r0:"
                  name="phoneNumber"
                  placeholder="Ex : 0548548539543"
                  type="number"
                />
              </div>
            </div>
            <div
              class="chakra-stack css-e8q362"
            >
              <div
                class="chakra-stack css-13203fq"
                style="margin: 0px;"
              >
                <p
                  class="chakra-text css-1bmwq82"
                >
                  City
                </p>
                <input
                  class="chakra-input css-1nkx82e"
                  id="field-:r0:"
                  name="city"
                  placeholder="Ex : Purwokerto"
                  type="text"
                />
              </div>
              <div
                class="chakra-stack css-13203fq"
                style="margin: 0px;"
              >
                <p
                  class="chakra-text css-1bmwq82"
                >
                  Districts
                </p>
                <input
                  class="chakra-input css-1nkx82e"
                  id="field-:r0:"
                  name="district"
                  placeholder="Ex : Banyumas"
                  type="text"
                />
              </div>
            </div>
            <div
              class="chakra-stack css-e8q362"
            >
              <div
                class="chakra-stack css-13203fq"
                style="margin: 0px;"
              >
                <p
                  class="chakra-text css-1bmwq82"
                >
                  Province
                </p>
                <input
                  class="chakra-input css-1nkx82e"
                  id="field-:r0:"
                  name="province"
                  placeholder="Ex : Cetral Java"
                  type="text"
                />
              </div>
              <div
                class="chakra-stack css-13203fq"
                style="margin: 0px;"
              >
                <p
                  class="chakra-text css-1bmwq82"
                >
                  Zip Code
                </p>
                <input
                  class="chakra-input css-1nkx82e"
                  id="field-:r0:"
                  name="zipCode"
                  placeholder="Ex : 67433"
                  type="number"
                />
              </div>
            </div>
            <div
              class="chakra-stack css-13203fq"
              style="margin: 0px;"
            >
              <p
                class="chakra-text css-1bmwq82"
              >
                Address
              </p>
              <textarea
                class="chakra-textarea css-xmuutn"
                id="field-:r0:"
                placeholder="Ex : Pramuka street number 49, Singapore"
                style="margin: 0px;"
              />
            </div>
          </div>
        </form>
      </div>
      <div
        class="chakra-stack css-ml5pvv"
      >
        <div
          class="chakra-button__group css-18fvkf4"
          data-orientation="horizontal"
          role="group"
        >
          <a
            href=""
          >
            <button
              class="chakra-button css-13qmsx6"
              type="button"
            >
              Next to Payment
            </button>
          </a>
          <a
            href="/"
            style="margin: 0px;"
          >
            <button
              class="chakra-button css-1u6isdo"
              type="button"
            >
              Back to Shopping Cart
            </button>
          </a>
        </div>
      </div>
    </div>
  </div>
  <span
    hidden=""
    id="__chakra_env"
  />
</div>
`;
